CMakeLists.txt

```diff
@@ -492,15 +491,186 @@ tr_add_external_auto_library(DEFLATE libdeflate ${DEFLATE_LIB_NAME}
         -DLIBDEFLATE_BUILD_SHARED_LIB=OFF
         -DLIBDEFLATE_BUILD_GZIP=OFF)
 
-tr_add_external_auto_library(EVENT2 libevent event
-    TARGET libevent::event
-    CMAKE_ARGS
-        -DEVENT__DISABLE_OPENSSL:BOOL=ON
-        -DEVENT__DISABLE_BENCHMARK:BOOL=ON
-        -DEVENT__DISABLE_TESTS:BOOL=ON
-        -DEVENT__DISABLE_REGRESS:BOOL=ON
-        -DEVENT__DISABLE_SAMPLES:BOOL=ON
-        -DEVENT__LIBRARY_TYPE:STRING=STATIC)
+if(CRYPTO_PKG STREQUAL "openssl")
+    add_compile_definitions(WITH_OPENSSL)
```

Comment:
Why are you [seemingly] duplicating functionality hidden behind `tr_add_external_auto_library()` function here, two times? Haven't compared what the difference between these two (if/else) branches and the original function is, but there must be a better way to do this. If it's a variation that can be parameterized, let's introduce a parameter for it instead.

Reply:
Here I have to link 2 libs (libevent and libevent_openssl) but `tr_add_external_auto_library()` only supports one, so any better way to implement this?

Reply:
parameter introduced